Sources/TuistGenerator/Generator/SchemeDescriptorsGenerator.swift

```diff
@@ -521,7 +521,7 @@ final class SchemeDescriptorsGenerator: SchemeDescriptorsGenerating {
                                             generatedProject: GeneratedProject,
                                             rootPath: AbsolutePath) -> RelativePath
     {
-        let xcodeProjectPath = targetNode.path.appending(component: generatedProject.name)
+        let xcodeProjectPath = targetNode.project.xcodeProjPath.appending(component: generatedProject.name)
```

Comment:
This may not be correct :/ (oddly testing locally seemed to have worked anyways 😅)

e.g. `/Projects/ProjectA/`

- `targetNode.path` == `/Projects/ProjectA/`
- `targetNode.project.path` == `/Projects/ProjectA/`
- `targetNode.project.xcodeProjPath` == `/Projects/ProjectA/ProjectA.xcodeproj` 
    - or in this case `/some/temp/dir/ProjectA.xcodeproj`

If I followed along correctly the issue is ensure paths within schemes respect the real `xcodeProjPath` which may can be in a different location like a temp directory. Good spot! indeed the logic in the scheme descriptor here needs some fixes :)

Testing the branch locally I believe you may be after:

`let xcodeProjectPath = targetNode.project.xcodeProjPath`